### PR TITLE
Blob DB: update blob file format

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Summary:
Changing blob file format and some code cleanup around the change. The change with blob log format are:
* Remove timestamp field in blob file header, blob file footer and blob records. The field is not being use and often confuse with expiration field.
* Blob file header now come with column family id, which always equal to default column family id. It leaves room for future support of column family.
* Compression field in blob file header now is a standalone byte (instead of compact encode with flags field)
* Blob file footer now come with its own crc.
* Key length now being uint64_t instead of uint32_t
* Blob CRC now checksum both key and value (instead of value only).
* Some reordering of the fields.

The list of cleanups:
* Better inline comments in blob_log_format.h
* rename ttlrange_t and snrange_t to ExpirationRange and SequenceRange respectively.
* simplify blob_db::Reader
* Move crc checking logic to inside blob_log_format.cc

Test Plan:
Existing tests

### CODE REVIEW

File: utilities/blob_db/blob_log_format.h [hunk 1]
```diff
@@ -9,243 +9,113 @@
...
-
-  tsrange_t GetTimeRange() const {
-    if (ts_range_) {
-      return *ts_range_;
-    }
-    return {0, 0};
-  }
-
-  const snrange_t& GetSNRange() const { return sn_range_; }
+// Format of blob log file footer (44 bytes):
```
Comment (sagar0) [inline utilities/blob_db/blob_log_format.h]: s/44/48.

File: utilities/blob_db/blob_db_test.cc [hunk 2]
```diff
@@ -256,6 +259,7 @@ TEST_F(BlobDBTest, PutUntil) {
   ASSERT_EQ(100 - data.size(), gc_stats.num_deletes);
   ASSERT_EQ(data.size(), gc_stats.num_relocate);
   VerifyDB(data);
+  ASSERT_FALSE(true);
```
Comment (sagar0) [inline utilities/blob_db/blob_db_test.cc]: Remove. Added while debugging I believe.

File: utilities/blob_db/blob_db_test.cc [hunk 3]
```diff
@@ -128,6 +128,9 @@ class BlobDBTest : public testing::Test {
     Iterator *iter = db->NewIterator(ReadOptions());
     iter->SeekToFirst();
     for (auto &p : data) {
+      if (!iter->Valid()) {
+        printf("s = %s\n", iter->status().ToString().c_str());
+      }
```
Comment (sagar0) [inline utilities/blob_db/blob_db_test.cc]: Added while debugging I believe.

File: utilities/blob_db/blob_file.h [hunk 4]
```diff
@@ -34,6 +37,10 @@ class BlobFile {
   // after that
   uint64_t file_number_;
 
+  // If true, the keys in this file all has TTL. Otherwise all keys don't
+  // have TTL.
+  bool has_ttl_;
```
Comment (sagar0) [inline utilities/blob_db/blob_file.h]: Then this can be used to determine if a blob file is a simple blob-file or not, isn't it?

**Comment by yiwu-arbug** [comment]
Apparently I was totally in a hurry last night :/

File: utilities/blob_db/blob_file.h (hunk 4 above)
Comment (yiwu-arbug) [reply utilities/blob_db/blob_file.h]: yes.

**Pull request closed without merging**